compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -1027,6 +1036,41 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       ddef
     }
   }
+
+  private def checkInternalAnnotations(tree: Tree): Unit = {
+    /* The compiler internal annotations are categorized into two:
+     * 1) postTyper: are annotations that are inserted in any phase
+     *   after the typer.
+     * 2) preTyper: are annotations that are inserted before typer
+     *   (such as @WasPublicBeforeTyper), as the type is required to find them,
+     *   the check must be after typer. We use an heuristic to find the
+     *   instances of these annotations based on the existence of their position
+     *   in the source file. Unfortunately it is still possible for macros to
+     *   generate such annotations trees without positions.
+     *
+     * WARNING: This check must be done in the earliest phase after typer that
+     * inserts scalajs complier internal annotations.
```

Comment:
What do you intend with this comment? If you put it, explain why (although it is kind of obvious). (I do not understand why someone would move the check though).